Source/Core/DiscIO/VolumeTri.h

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+#include "Common/Lazy.h"
+#include "DiscIO/Filesystem.h"
+#include "DiscIO/Volume.h"
+#include "DiscIO/VolumeDisc.h"
+
+namespace DiscIO
+{
+class BlobReader;
+enum class BlobType;
+enum class Country;
+class FileSystem;
+enum class Language;
+enum class Region;
+enum class Platform;
+
+class VolumeTri final : public VolumeDisc
+{
+public:
+  explicit VolumeTri(std::unique_ptr<BlobReader> reader);
+  ~VolumeTri() override;
+  bool Read(u64 offset, u64 length, u8* buffer,
+            const Partition& partition = PARTITION_NONE) const override;
+  const FileSystem* GetFileSystem(const Partition& partition = PARTITION_NONE) const override;
+  std::string GetGameTDBID(const Partition& partition = PARTITION_NONE) const override;
+  std::map<Language, std::string> GetShortNames() const override;
+  std::map<Language, std::string> GetLongNames() const override;
+  std::map<Language, std::string> GetShortMakers() const override;
+  std::map<Language, std::string> GetLongMakers() const override;
+  std::map<Language, std::string> GetDescriptions() const override;
+  std::vector<u32> GetBanner(u32* width, u32* height) const override;
+
+  Platform GetVolumeType() const override;
+  bool IsDatelDisc() const override;
+  Region GetRegion() const override;
+  BlobType GetBlobType() const override;
+  u64 GetSize() const override;
+  bool IsSizeAccurate() const override;
+  u64 GetRawSize() const override;
+  const BlobReader& GetBlobReader() const override;
+
+  std::array<u8, 20> GetSyncHash() const override;
+
+private:
+  static const u32 GC_BANNER_WIDTH = 96;
+  static const u32 GC_BANNER_HEIGHT = 32;
```

Comment:
Any reason we don't make those `constexpr`?

Reply:
VS complains about `constexpr` not being valid here when I replace them, and it messes up the functions below